--- emsdk-orig/upstream/emscripten/system/lib/html5/callback.c	2026-07-23 19:19:44.000000000 -0400
+++ emsdk/upstream/emscripten/system/lib/html5/callback.c	2026-07-29 22:22:52.902785632 -0400
@@ -16,7 +16,7 @@
   event_callback callback;
   int event_type;
   void *user_data;
-  uint8_t event_data[];
+  __attribute__((aligned(8))) uint8_t event_data[];
 } callback_args_t;
 
 static void do_callback(void* arg) {
